packages/core/logger/src/Logger.js

```diff
@@ -2,27 +2,18 @@
 
 import type {IDisposable, LogEvent} from '@parcel/types';
 
-import EventEmitter from 'events';
+import {ValueEmitter} from '@parcel/events';
 import {inspect} from 'util';
 
 class Logger {
-  // TODO: This can't be explicitly annotated as an EventEmitter since
-  // declared private properties with type annotations break eslint's
-  // no-unused-var rule (even with babel-eslint). Annotate this when
-  // things aren't broken: https://github.com/babel/babel-eslint/issues/688
-  #emitter = new EventEmitter();
+  _logEmitter: ValueEmitter<LogEvent> = new ValueEmitter();
 
   onLog(cb: (event: LogEvent) => mixed): IDisposable {
-    this.#emitter.addListener('log', cb);
-    return {
-      dispose: () => {
-        this.#emitter.removeListener('log', cb);
-      }
-    };
+    return this._logEmitter.addListener(cb);
   }
 
   verbose(message: string): void {
-    this.#emitter.emit('log', {
+    this._logEmitter.emit({
```

Comment:
not sure if we should deviate from the standard eventemitter model that's part of Node.js.

Perhaps this can still remain as `emit(type, value)` ?

It would probably also make some stuff easier and possible faster for event listening? (if the listener only cares about 1 type)

Reply:
> not sure if we should deviate from the standard eventemitter model that's part of Node.js.

I hear you. This implements the semantics of EventEmitter nearly identically, but with a slightly different API to help us better type things. The model is the same, except instead of registering arbitrary misspellable strings for `any`-typed callbacks there's a concrete, typed API and internal structure.

> Perhaps this can still remain as emit(type, value) ?

> It would probably also make some stuff easier and possible faster for event listening? (if the listener only cares about 1 type)

I should have provided a better example than the logger :P It's not that there's only one emitter with a single list of handlers, pushing the responsibility of filtering events to the caller.

Instead, the intent is that you create one value emitter for each type of "event" you'd use an EventEmitter for, along with a public method for registering listeners. 

This way, the "type" is encoded as part API and itself can have documentation, type annotation, etc.

Let's say in addition to wanting to observe logs, we wanted to observe "foo" events (like you would with `logger.on('foo', ...)`:

```js
class Logger {
  _logEmitter: ValueEmitter<LogEvent> = new ValueEmitter();
  _fooEmitter: ValueEmitter<Foo> = new ValueEmitter();

  onLog(cb: (event: LogEvent) => mixed): IDisposable {
    return this._logEmitter.addListener(cb);
  }
  
  onFoo(cb: (event: Foo) => mixed): IDisposable {
    return this._fooEmitter.addListener(cb);
  }
```

... and then you could emit Foos by calling `this._fooEmitter.emit(myFoo)` within the logger.

Re: easier -- I can totally relate to this too. Creating `n` emitters along with `n` public api methods for `n` "events" seems like a lot. But this way these events are very clearly part of the public api and can be easily documented and type checked. If we wanted to type an event emitter like in the [alternate suggestion](https://stackoverflow.com/questions/49411515/limiting-the-type-of-eventname-in-a-class-that-extends-eventemitter-with-flow), it would probably look something like this (I've never actually done/tested this; just riffing on that SO post):

```js
class Logger extends EventEmitter {
	on('log', (log: Log) => mixed)): this;
	on('foo', (foo: Foo) => mixed)): this;
	on(event: string, ...args: Array<any>): this;
	emit('log', log: Log): this;
	emit('foo', foo: Foo): this;
	emit(event: string, ...args: Array<any>): this;
}
```

...which not only loses the typesafety in the misspelled-event-name case, since it must at minimum satisfy the `EventEmitter` interface, but IMHO is just as tedious and verbose.